src/queries/catalog/resolve-value-query/resolve-value-query-executor.ts

```diff
@@ -0,0 +1,43 @@
+import type { ResolveValueQuery, ResolveValueQueryResult } from './resolve-value-query-format';
+import { log } from '../../../util/log';
+import type { BasicQueryData } from '../../base-query-format';
+import { slicingCriterionToId } from '../../../slicing/criterion/parse';
+import { resolveToValues } from '../../../dataflow/environments/resolve-by-name';
+import { recoverName } from '../../../r-bridge/lang-4.x/ast/model/processing/node-id';
+
+export function fingerPrintOfQuery(query: ResolveValueQuery): string {
+	return JSON.stringify(query);
+}
+
+export function executeResolveValueQuery({ dataflow: { graph, environment } }: BasicQueryData, queries: readonly ResolveValueQuery[]): ResolveValueQueryResult {
+	const idMap = graph.idMap;
+
+	if(!idMap) {
+		throw new Error('idMap was undefined');
+	}
```

Comment:
Please use flowR's `guard` function for this, to ensure that the idMap is not undefined.
In general, you can always fallback to the idMap given in then normalized AST (as part of the BasicQueryData). So we should do this first.